src/providers/wms/qgswmsprovider.cpp

```diff
@@ -276,7 +276,12 @@ QgsWmsProvider::QgsWmsProvider( QString const &uri, const ProviderOptions &optio
 QString QgsWmsProvider::prepareUri( QString uri )
 {
   // some services provide a percent/url encoded (legend) uri string, always decode here
-  uri = QUrl::fromPercentEncoding( uri.toUtf8() );
+  if ( uri.startsWith( "https%3A%2F%2F" ) || uri.startsWith( "http%3A%2F%2F" ) )
+  {
+    qDebug() << "========================================================================= QgsWmsProvider::prepareUri" << uri;
+    uri = QUrl::fromPercentEncoding( uri.toUtf8() );
+    Q_ASSERT( false );
+  }
```

Comment:
@cioddi I do not find a unit test or a user case using this code. Does it still valid?

Reply:
Should be reached through the QgsWmsProvider constructor.

Reply:
Ok I will revert this section and squash some commit to improve the git history.

Reply:
done!